### ISSUE TITLE
Compiling esm esm cannot be used directly in the bundler module

### ISSUE THREAD

**Issue by Mrxyy** [AssemblyScript/assemblyscript#2659]
### Bug description

Compiling esm code cannot be used directly in the bundler module，.e.g:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/49010414/221726549-6a8ac4a7-1fdf-4602-b6f8-9cd9550ab203.png">
The Object.assign cant work. Because bundle module is not the same as browser esm.module export variable just only have getter.
<img width="710" alt="image" src="https://user-images.githubusercontent.com/49010414/221727203-ffc6806d-3353-437e-ada4-328046ea102c.png">
![image](https://user-images.githubusercontent.com/49010414/221727362-72005d78-b24a-40ff-8cea-353f117947bf.png)

### Steps to reproduce

1. use binding `@external("../test","test")`
2. complie esm code
<img width="351" alt="image" src="https://user-images.githubusercontent.com/49010414/221727811-204c1038-edd3-4b68-879f-e221ee1ff90a.png">
3. I hava a Compatibility method,but i think Not good enough.
<img width="703" alt="image" src="https://user-images.githubusercontent.com/49010414/221728288-94dd3734-9de0-494c-b9e2-d9b9f0f84855.png">

### AssemblyScript version

^0.27.0

**Comment by CountBleck** [comment]
What bundler are you using?

**Comment by Mrxyy** [comment]
> What bundler are you using?

webpack5

**Comment by CountBleck** [comment]
I suppose the solution is to use `Object.defineProperties()` instead of `Object.assign()`

**Comment by Mrxyy** [comment]
> . I hava a Compatibility method,but i think Not good enough.

Yes, the above code is just to ensure that the current compiled results work.

**Comment by Mrxyy** [comment]
> I suppose the solution is to use `Object.defineProperties()` instead of `Object.assign()`

Yes, the above code is just to ensure that the current compiled results work.

**Comment by YSandro** [comment]
Same problem.
Generated binding to esm:
```
const __module0 = imports["./main.js"];
const adaptedImports = {
  "./main.js": Object.assign(Object.create(__module0), {
    myFunc(x) {
      // assembly/index/my_func(i32) => bool
      return __module0.myFunc(x) ? 1 : 0;
    },
  }),
};
```
In Chrome and Vivaldi works fine.
if Firefox throw:
```
Uncaught TypeError: "myFunc" is read-only
```
If I edit manually to
```
const __module0 = imports["./main.js"];
const adaptedImports = {
  "./main.js": Object.assign({
    myFunc(x) {
      // assembly/index/my_func(i32) => bool
      return __module0.myFunc(x) ? 1 : 0;
    },
  }),
};
```
or to
```
const __module0 = imports["./main.js"];
  "./main.js": Object.create(__module0),
};
```
it works.
Also works when calling func via globalThis in env.ts:

```
// env
@external("env", "myFunc")
export declare function my_func(x: f32): boolean
```
main.ts:
```
// ./main.js
globalThis.myFunc = (x: number): boolean => {
  console.info('main.js myFunc()');
  return true;
}
```
But manual edits or calls through a global object are unnecessary work.
